libs/community/langchain_community/tools/azure_ai_services/azure_file_translation.py

```diff
@@ -0,0 +1,162 @@
+from __future__ import annotations
+
+import logging
+import os
+from typing import Any, Dict, Optional
+
+from langchain_core.callbacks import CallbackManagerForToolRun
+from langchain_core.tools import BaseTool
+from langchain_core.utils import get_from_dict_or_env
+
+from langchain_community.document_loaders import (
+    UnstructuredExcelLoader,
+    UnstructuredHTMLLoader,
+    UnstructuredPDFLoader,
+    UnstructuredPowerPointLoader,
+    UnstructuredWordDocumentLoader,
+    UnstructuredXMLLoader,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class AzureFileTranslateTool(BaseTool):
+    """
+    A tool that uses Azure Text Translation API to translate a text document from
+    any language into a target language.
+    """
+
+    text_translation_key: str = ""
+    text_translation_endpoint: str = ""
+    target_language: str = "en"
+    translate_client: Any
+
+    name: str = "azure_document_translation"
+    description: str = """
+        A Wrapper around Azure AI Services can be used to
+        translate a document into a specific language.
+        It reads the text from a file, processes it,
+        and then outputs with the desired language.
+        """
+
+    @classmethod
+    def validate_environment(cls, values: Dict) -> Any:
```

Comment:
Missing model_validator (you can import it from `from pydantic import model_validator`)

```suggestion

    @model_validator(mode="before")
    @classmethod
    def validate_environment(cls, values: Dict) -> Any:
```